src/Symfony/Component/Security/Core/User/User.php

```diff
@@ -142,6 +142,13 @@ public function isEqualTo(UserInterface $user)
             return false;
         }
 
+        $currentRoles = array_map('strval', (array) $this->getRoles());
+        $newRoles = array_map('strval', (array) $user->getRoles());
+        $rolesChanged = \count($currentRoles) !== \count($newRoles) || \count($currentRoles) !== \count(array_intersect($currentRoles, $newRoles));
```

Comment:
Can we simplify to `$currentRoles != $newRoles`?

Reply:
It won’t work because we cannot guarantee order of roles.

‘’’
[0,1]==[1,0] // false
‘’’